kafka-rest/src/test/java/io/confluent/kafkarest/resources/v3/ListPartitionOffsetsActionTest.java

```diff
@@ -0,0 +1,218 @@
+          TOPIC_NAME,
+          /* partitionId= */ 1,
+          Arrays.asList(
+              PartitionReplica.create(
+                  CLUSTER_ID,
+                  TOPIC_NAME,
+                  /* partitionId= */ 1,
+                  /* brokerId= */ 2,
+                  /* isLeader= */ true,
+                  /* isInSync= */ false),
+              PartitionReplica.create(
+                  CLUSTER_ID,
+                  TOPIC_NAME,
+                  /* partitionId= */ 1,
+                  /* brokerId= */ 3,
+                  /* isLeader= */ false,
+                  /* isInSync= */ true),
+              PartitionReplica.create(
+                  CLUSTER_ID,
+                  TOPIC_NAME,
+                  /* partitionId= */ 1,
+                  /* brokerId= */ 1,
+                  /* isLeader= */ false,
+                  /* isInSync= */ false)));
+  private static final Partition PARTITION_3 =
+      Partition.create(
+          CLUSTER_ID,
+          TOPIC_NAME,
+          /* partitionId= */ 2,
+          Arrays.asList(
+              PartitionReplica.create(
+                  CLUSTER_ID,
+                  TOPIC_NAME,
+                  /* partitionId= */ 2,
+                  /* brokerId= */ 3,
+                  /* isLeader= */ true,
+                  /* isInSync= */ false),
+              PartitionReplica.create(
+                  CLUSTER_ID,
+                  TOPIC_NAME,
+                  /* partitionId= */ 2,
+                  /* brokerId= */ 1,
+                  /* isLeader= */ false,
+                  /* isInSync= */ true),
+              PartitionReplica.create(
+                  CLUSTER_ID,
+                  TOPIC_NAME,
+                  /* partitionId= */ 2,
+                  /* brokerId= */ 2,
+                  /* isLeader= */ false,
+                  /* isInSync= */ false)));
+
+  private static final Topic TOPIC =
+      Topic.create(
+          CLUSTER_ID,
+          TOPIC_NAME,
+          Arrays.asList(PARTITION_1, PARTITION_2, PARTITION_3),
+          /* replicationFactor= */ (short) 3,
+          /* isInternal= */ false,
+          /* authorizedOperations= */ emptySet());
+  @Mock private Admin adminClient;
+  @Mock private TopicManager topicManager;
+  @Mock private PartitionManager partitionManager;
+
+  @Mock private ListOffsetsResult earliestResult;
+
+  @Mock private ListOffsetsResult.ListOffsetsResultInfo earliestResultInfo1;
+
+  @Mock private ListOffsetsResult latestResult;
+
+  @Mock private ListOffsetsResult.ListOffsetsResultInfo latestResultInfo1;
+
+  private ListPartitionOffsetsAction listPartitionOffsetsAction;
+
+  @BeforeEach
+  public void setUp() {
+    listPartitionOffsetsAction =
+        new ListPartitionOffsetsAction(
+            () -> partitionManager,
+            new CrnFactoryImpl(/* crnAuthorityConfig= */ ""),
+            new FakeUrlFactory());
+  }
+
+  @Test
+  public void listPartitionOffsets_existingPartitionWithOffsets_returnsPartitionWithOffsets() {
```

Comment:
can you add the negative test case for the following? And see the API returns appropriate response.
* cluster is not found
* topic is not found
* partition is not found